### PR TITLE
Improvement for drop on repositories view + test project for greg

### PR DESCRIPTION
**Pull request by pkriens**

### CODE REVIEW

File: bndtools.test/bndws/test.debug.dep/src/test/debug/dep/Foo.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package test.debug.dep;
+
+public class Foo {
+	{
+		System.out.println("Foo created");
```
Comment (gamerson) [inline bndtools.test/bndws/test.debug.dep/src/test/debug/dep/Foo.java]: @pkriens  Is it here where you set the breakpoint and eclipse can't load it?

File: bndtools.test/bndws/test.debug/src/test/bottom/TopTest.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package test.bottom;
+
+import test.debug.dep.Foo;
+import junit.framework.TestCase;
+
+public class TopTest extends TestCase {
+
+	public void testX() {
+		new Foo();
+		System.out.println("hello");
```
Comment (gamerson) [inline bndtools.test/bndws/test.debug/src/test/bottom/TopTest.java]: or here?

**Comment by pkriens** [comment]
I set a breakpoint at setX and then single step. So testX is loaded correctly but then Foo.java is not editable. Thanks for working on this!

**Comment by gamerson** [comment]
so Foo.java is loaded from OSGI-OPT/src and thus uneditable instead of being loaded from the real java project?

**Comment by pkriens** [comment]
Yup … for some reason this has priority.

**Comment by gamerson** [comment]
I remember when this change happened in 3.x dev.  It was during some improvements to "real classpath" visibility made in BND container awhile back where the generated jar was listed first on the classpath before the "eclipse project reference".  I recall this was on purpose but I don't remember the exact reason. 

cc @bjhargrave

**Comment by pkriens** [comment]
Didn’t Neil do some work? Any chance we can get into the heart of the source selector? Would be great if we could skip the project on the container.

**Comment by gamerson** [comment]
I've found the cause of this.  It is a little involved and has a few moving parts (project .classpath config and also bnd.bnd -buildpath config) so I will move that discussion to here https://github.com/bndtools/bndtools/issues/1395

**Comment by gamerson** [comment]
@pkriens quick fix for you, in the test.debug.dep/.classpath, open that file, edit the first entry change it from

`<classpathentry kind="src" output="bin" path="src"/>`
to
`<classpathentry kind="src" path="src"/>`

Delete any existing JUnit launches and then relaunch JUnit.  That should make things work.

**Pull request merged**